### PR TITLE
Issue 24160 dot favorite add favorite page to context menu

### LINKED ISSUE #24160: UI - Favorite Page portlet: add / edit favorite page from context menu
### Parent Issue
#22343 

### Task
As an user I want to be able to Create/Edit a favorite page directly from the listing table context menu (currently to create a new favorite page a user needs to go first to EDIT MODE and click on the star icon to create it)

image

### Proposed Objective

Please Select

### Proposed Priority

Please Select

### Acceptance Criteria

- On the context menu a new option for Add/Edit Favorite will be added
- if option selected, then the ADD/EDIT Favorite dialog will be shown with the page data

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
As an user I want to be able to Create/Edit a favorite page directly from the listing table context menu (currently to create a new favorite page a user needs to go first to EDIT MODE and click on the star icon to create it)

image

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
- On the context menu a new option for `Add Favorite` if no favorite exist on the page, and `Edit Favorite` if a favorite already exist
- if option selected, then the ADD/EDIT Favorite dialog will be shown with the page data

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts [hunk 1]
```diff
@@ -578,25 +590,63 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
         });
     };
 
-    private getFavoritePagesData = (limit: number, identifier?: string) => {
-        const identifierQuery = identifier ? `+identifier:${identifier}` : '';
+    private getFavoritePagesData = (limit: number, identifier?: string, url?: string) => {
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:593]: Maybe the parameters of this function could be one object

**Review by zJaaal — COMMENTED** [review]
.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 1 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:593]: Totally.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 1 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts]: done!

**Review by zJaaal — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Pull request merged**